specification/quantum/Quantum.Workspace/operations/sessions.tsp

```diff
@@ -82,16 +84,28 @@ namespace Quantum.Workspace.Operations {
     get is GetOperation<SessionDetails>;
 
     /** List all Sessions. */
-    list is ListOperation<SessionDetails>;
+    #suppress "@azure-tools/typespec-azure-core/use-standard-operations" """
+      We had to use a custom ResourceList to avoid an autoroute bug for workspace/jobs and sesssions/{sessionId}/jobs.
+      See https://github.com/Azure/typespec-azure/issues/494
+      """
+    list is CustomListOperation<
+      Resource = SessionDetails,
+      Response = PagedSessionDetails,
+      Parameters = {
+        ...CustomFilterQueryParameter;
+        ...CustomSkipQueryParameter;
+        ...CustomTopQueryParameter;
+        ...CustomOrderByQueryParameter;
+      }
+    >;
 
     /** List jobs in a session. */
-    #suppress "@azure-tools/typespec-azure-core/no-operation-id" ""
     #suppress "@azure-tools/typespec-azure-core/no-openapi" "The nesting of Jobs under a Session makes the auto-generated ID to be Sessions_JobsList, which is not in a Noun_Verb format."
+    #suppress "@azure-tools/typespec-azure-core/use-standard-names" "Existing name"
     #suppress "@azure-tools/typespec-azure-core/use-standard-operations" """
       We had to use a custom ResourceList to avoid an autoroute bug for workspace/jobs and sesssions/{sessionId}/jobs.
       See https://github.com/Azure/typespec-azure/issues/494
       """
-    #suppress "@azure-tools/typespec-azure-core/use-standard-names" "Existing name"
     @operationId("SessionsJobs_List")
```

Comment:
@allenjzhang If I remember correctly, use of `@operationId` is discouraged?

Reply:
Correct. It is discouraged. However in Quantum cases they have multiple parents situation that we have not solved. It does provide an escape valve. So I can sign off.